scripts/single-node-all-upgrades.sh

```diff
@@ -107,21 +105,12 @@ startCelestiaApp() {
     --force-no-bbr
 }
 
-upgradeToV3AndV4() {
-    sleep 30
-    echo "Waiting for app version 2 before proceeding..."
-    while true; do
-        current_version=$(celestia-appd status | jq -r '.node_info.protocol_version.app')
-        if [ "$current_version" = "2" ]; then
-            echo "App version 2 detected, proceeding with v3 upgrade..."
-            break
-        fi
-        echo "Current version: $current_version, waiting for version 2..."
-        sleep 1
-    done
+# Function to perform upgrade to a specific version
+performUpgrade() {
```

Comment:
is it expected that the user running this script does some sanity checks after the upgrade succeeds? 

there's no query based assertions built into the script, right?